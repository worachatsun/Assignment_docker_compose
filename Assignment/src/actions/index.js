export * from './PreferenceActions.js'
export * from './AuthActions.js'