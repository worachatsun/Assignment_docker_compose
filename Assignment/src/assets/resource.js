const languages = [
    'English', 'Afar', 'Abkhazian' ,'Afrikaans' ,
    'Amharic' ,'Arabic' ,'Assamese' ,'Aymara' , 
    'Azerbaijani' ,'Bashkir' ,'Byelorussian' ,'Bulgarian' , 
    'Bihari' ,'Bislama'
]

const timezones = [
    '(UTC+00:00) El Aaiun', '(UTC+00:00) Faroe', '(UTC+00:00) Freetown',
    '(UTC+00:00) Guernsey', '(UTC+00:00) Isle of Man', '(UTC+00:00) Jersey',
    '(UTC+00:00) Lisbon', '(UTC+00:00) Lome', '(UTC+00:00) London',
    '(UTC+00:00) Madeira', '(UTC+00:00) Monrovia', '(UTC+00:00) Nouakchott',
    '(UTC+00:00) Ouagadougou', '(UTC+00:00) Reykjavik', '(UTC+00:00) Sao Tome',
    '(UTC+00:00) St. Helena', '(UTC+00:00) UTC', '(UTC+01:00) Algiers',
    '(UTC+01:00) Amsterdam', '(UTC+01:00) Andorra', '(UTC+01:00) Bangui',
    '(UTC+01:00) Belgrade', '(UTC+01:00) Berlin', '(UTC+01:00) Bratislava'
]

const currencies = [
    'USD ($)', 'EUR (€)', 'GBP (£)', 'RUB (pуб)', 'HKD ($)',
    'JPY (yen)', 'NOK (kr)', 'TWD ($)', 'THB (฿)', 	'TRY (₺)'
]

export {
    languages, timezones, currencies
}