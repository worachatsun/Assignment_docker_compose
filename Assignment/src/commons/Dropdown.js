import React, { Component } from 'react'
import styled from 'styled-components'
import { FaCaretDown, FaCaretUp } from 'react-icons/lib/fa';

export default class Dropdown extends Component {
    constructor(props) {
        super(props)
        this.state = {
            dropdownActive: false,
            dropdownValue: this.props.displayValue
        }
        this.props.setData(this.props.value[0])
    }

    setDropdownValue(value) {
        this.setState({
            dropdownValue: value,
            dropdownActive: !this.state.dropdownActive
        })
        this.props.setData(value)
    }

    render() {
        return (
            <DropdownContainer>
                <div onClick={() => this.setState({dropdownActive: !this.state.dropdownActive})}>
                    <DropdownContent>
                        {this.props.displayValue}{this.state.dropdownActive?<FaCaretUp/>:<FaCaretDown/>}
                    </DropdownContent>
                </div>
                {this.state.dropdownActive?menu(this.props.value, this.setDropdownValue.bind(this)):''}
            </DropdownContainer>
        )
    }
}

const menu = (data = [], setDropdownValue) => (
    <DropdownList>
        {data.map((value, index) => <DropdownContent onClick={() => setDropdownValue(value)} key={index}>{value}</DropdownContent>)}
    </DropdownList>
)

const DropdownContainer = styled.div`
    width: 100%;
    background-color: #F8F8F8;
    border: 1px solid #D1D5D7;
    border-radius: 2px;
    margin: 5px 0 13px 0;
`
const DropdownContent = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 4px 0 4px;
    font-size: 0.9em;
    &:hover {
        background-color: #ddd;
    }
`

const DropdownList = styled.div`
    background-color: #F8F8F8;
    position: absolute;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
    max-height: 160px;
    padding: 12px 16px;
    overflow-y: scroll;
`