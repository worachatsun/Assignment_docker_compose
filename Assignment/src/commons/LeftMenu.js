import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { MdLockOutline, MdLocalShipping, MdAccountCircle, MdGridOn, MdLocalAtm, MdNotifications, MdAssignment } from 'react-icons/lib/md'
import { FaCreditCard } from 'react-icons/lib/fa'
import { TiClipboard } from 'react-icons/lib/ti'

export default class LeftMenu extends Component {
    render() {
        return (
            <Container>
                <MenuButton><MdAccountCircle size={20} style={{marginRight: 8}}/>Edit Profile</MenuButton>
                <MenuButton active={this.props.nowPath==='/'?true:false}><MdAssignment size={20} style={{marginRight: 8}} /><Link to="/" style={{textDecoration: 'none', color: '#6F727D'}}>Preferences</Link></MenuButton>
                <MenuButton><MdLockOutline size={20} style={{marginRight: 8}}/>Password</MenuButton>
                <MenuButton><MdNotifications size={20} style={{marginRight: 8}} />Notifications</MenuButton>
                <MenuButton><MdGridOn size={20} style={{marginRight: 8}} />Connected Accounts</MenuButton>
                <SepSection/>
                <MenuButton><TiClipboard size={24} style={{marginRight: 8}} />Orders</MenuButton>
                <MenuButton><FaCreditCard size={20} style={{marginRight: 8}} />Payment</MenuButton>
                <MenuButton><MdLocalShipping size={20} style={{marginRight: 8}} />Shipping</MenuButton>
                <SepSection/>
                <MenuButton><MdLocalAtm size={20} style={{marginRight: 8}} />Credits & Referrals</MenuButton>
            </Container>
        )
    }
}

const Container = styled.div`
    width: 100%;
    color: #6F727D;
    border-radius: 2px;
    background-color: white;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
`

const MenuButton = styled.div`
    display: flex;
    flex: 1;
    padding: 5px 0 3px 15px;
    align-items: center;
    height: 30px;
    font-size: 14px;
    background-color: ${props => props.active?'#ddd':'white'};
    font-weight: ${props => props.active?'bold':'regular'};
    &:hover {
        font-weight: bold;
        background-color: #ddd;
    }
`

const SepSection = styled.div`
    border-bottom: 1px solid #ddd;
    margin: 0 10px 0 10px;
`