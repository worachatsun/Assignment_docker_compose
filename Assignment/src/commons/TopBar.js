import React, { Component } from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { signOut } from '../actions'
import { FaCaretDown, FaUser, FaShoppingCart, FaInbox, FaBolt, FaSignOut } from 'react-icons/lib/fa';

class TopBar extends Component{
    constructor(props){
        super(props)
        this.state = {
            dropdown: false
        }
    }

    render() {
        return (
            <HeaderDiv>
                <div>
                    {this.props.noSideBar?
                        ''
                    :
                        <SearchInput placeholder={'Search Fancy'} />
                    }
                </div>
                <HeaderText>
                    FANCY
                </HeaderText>
                <div>
                    {this.props.noSideBar?
                        ''
                    :
                        <IconGroup>
                            <FaShoppingCart fill={'#92979D'} />
                            <FaInbox fill={'#92979D'} />
                            <FaBolt fill={'#92979D'} />
                            <FaUser fill={'#92979D'} />
                            <span style={{color: '#92979D', marginTop: 3, fontSize: 14, fontWeight: 'bold'}}>You</span>
                            <Dropdown onClick={() => this.setState({dropdown: !this.state.dropdown})}>
                                <FaCaretDown fill={'#92979D'}/>
                                {this.state.dropdown?
                                    <DropdownContent>
                                        <div onClick={() => this.props.signOut()}><FaSignOut style={{marginRight: 10}}/>Sign out</div>
                                    </DropdownContent>
                                :
                                    ''
                                }
                            </Dropdown>
                        </IconGroup>
                    }
                </div>
            </HeaderDiv>
        )
    }
}

export default connect(null, { signOut })(TopBar)

const MakeRowDiv = styled.div`
    display: flex;
    flex: 1;
    flex-direction: row;
    justify-content: space-around;
`

const IconGroup = MakeRowDiv.extend`
    align-items: center;
    width: 160px;
`

const HeaderDiv = MakeRowDiv.extend`
    align-items: center;
    background-color: white;
    height: 48px;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
`

const SearchInput = styled.input`
    height: 24px;
    width: 170px;
    border-radius: 2px;
    border: 1px solid #ddd;
    background-color: #f2f2f2;
`

const HeaderText = styled.div`
    font-family: 'Quicksand';
    font-size: 1.5em;
    font-weight: bold;
    color: #515C67;
`

const Dropdown = styled.div`
     position: relative;
     display: inline-block;
 `

 const DropdownContent = styled.div`
     position: absolute;
     background-color: #f9f9f9;
     min-width: 100px;
     box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
     padding: 12px 16px;
     z-index: 1;
 `
