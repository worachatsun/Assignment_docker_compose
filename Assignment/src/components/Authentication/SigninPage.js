import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import TopBar from '../../commons/TopBar'
import { connect } from 'react-redux'
import { signIn } from '../../actions'

class SigninPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
            username: '',
            password: '',
            error: ''
        }
    }

    handleSubmit = () => {
        if(this.state.username!==''&this.state.password!=='')
            this.props.signIn(this.state)
        else
            this.setState({error: 'Please insert Username or Password'})
    }

    render() {
        return (
            <div>
                <TopBar noSideBar/>
                <OuterContainer>
                    <Container>
                        <div style={{fontWeight: 'bold', marginTop: 20, marginLeft: '2.3%', color: '#515C67'}}>Signin</div>
                        <SepSection />
                        <OuterContainer>
                            <Text>Username</Text>
                            <Input type="text" placeholder={'Username or Email'} onChange={e => this.setState({username: e.target.value})}/>
                        </OuterContainer> 
                        <OuterContainer>
                            <Text>Password</Text>
                            <Input type="password" placeholder={'Password'} onChange={e => this.setState({password: e.target.value})}/>
                        </OuterContainer>
                        <OuterContainer>
                            <Text color={'red'}>{this.state.error!==''?this.state.error:''}</Text>
                        </OuterContainer>
                        <SepSection />
                        <OuterContainer>
                            <Button onClick={() => this.handleSubmit()}>Signin</Button>
                            <Link to="signup"><Button>Signup</Button></Link>
                        </OuterContainer>
                    </Container>
                </OuterContainer>
            </div>
        )
    }
}

const OuterContainer = styled.div`
    display: flex;
    flex: 1;
    justify-content: center;
    margin-top: 15px;
`

const Container = styled.div`
    width: 50%;
    margin-top: 10%;
    background-color: white;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
`
const SepSection = styled.div`
    margin: 15px 0 15px 0;
    border-bottom: 1px solid #ddd;
`

const Text = styled.div`
    color: ${props => props.color || '#6F727D'};
    font-size: 1em;
`

const Button = styled.button`
    height: 34px;
    width: 200px;
    border-radius: 4px;
    margin: 0 15px 15px 15px;
    border: 1px solid #515C67;
    color: #515C67;
    font-size: 0.8em;
    font-family: 'Hind';
    font-weight: bold;
    background-color: '#F8F8F8';
    &:hover {
        font-weight: bold;
        background-color: #ddd;
    }
`
const Input = styled.input`
    border: 1px solid #ddd;
    border-radius: 3px;
    margin: 0 10px 0 10px;
`

export default connect(null, { signIn })(SigninPage)