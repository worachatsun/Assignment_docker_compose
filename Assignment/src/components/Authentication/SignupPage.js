import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import TopBar from '../../commons/TopBar'
import { connect } from 'react-redux'
import { signUp } from '../../actions'

class SignupPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
            username: '',
            email: '',
            password: '',
            error: ''
        }
    }

    handleSubmit = () => {
        if(this.state.username!==''&this.state.email!==''&this.state.password!=='')
            this.props.signUp(this.state)
        else
            this.setState({error: 'Something went wrong on Username, Email or Password'})
    }

    render() {
        return (
            <div>
                <TopBar/>
                <OuterContainer>
                    <Container>
                        <div style={{fontWeight: 'bold', marginTop: 20, marginLeft: '2.3%', color: '#515C67'}}>Signup</div>
                        <SepSection />
                        <OuterContainer>
                            <Text>Username</Text>
                            <Input type="text" placeholder={'Username'} onChange={e => this.setState({username: e.target.value})}/>
                            <Text>Email</Text>
                            <Input type="email" placeholder={'Email'} onChange={e => this.setState({email: e.target.value})}/>
                            <Text>Password</Text>
                            <Input type="password" placeholder={'Password'} onChange={e => this.setState({password: e.target.value})}/>
                        </OuterContainer>
                        <OuterContainer>
                            <Text color={'red'}>{this.state.error!==''?this.state.error:''}</Text>
                        </OuterContainer>
                        <SepSection />
                        <OuterContainer>
                            <Link to="signin"><Button>Signin</Button></Link>
                            <Button onClick={() => this.handleSubmit()}>Signup</Button>
                        </OuterContainer>
                    </Container>
                </OuterContainer>
            </div>
        )
    }
}

const OuterContainer = styled.div`
    display: flex;
    flex: 1;
    justify-content: center;
    margin-top: 15px;
`

const Container = styled.div`
    width: 50%;
    margin-top: 10%;
    background-color: white;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
`
const SepSection = styled.div`
    margin: 15px 0 15px 0;
    border-bottom: 1px solid #ddd;
`

const Text = styled.div`
    color: ${props => props.color || '#6F727D'};
    font-size: 1em;
`

const Button = styled.button`
    height: 34px;
    width: 200px;
    border-radius: 4px;
    margin: 0 15px 15px 15px;
    border: 1px solid #515C67;
    color: #515C67;
    font-size: 0.8em;
    font-family: 'Hind';
    font-weight: bold;
    background-color: '#F8F8F8';
    &:hover {
        font-weight: bold;
        background-color: #ddd;
    }
`
const Input = styled.input`
    border: 1px solid #ddd;
    border-radius: 3px;
    margin: 0 10px 0 10px;
`
export default connect(null, { signUp })(SignupPage)