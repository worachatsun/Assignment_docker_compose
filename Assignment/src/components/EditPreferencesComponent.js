import React, { Component } from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { createPreference, getPreference, deletePreference, updatePreference } from '../actions'
import TopBar from '../commons/TopBar'
import LeftMenu from '../commons/LeftMenu'
import { FaLock } from 'react-icons/lib/fa'
import Dropdown from '../commons/Dropdown'
import { languages, timezones, currencies } from '../assets/resource'

class EditPreferencesComponent extends Component {
    constructor(props) {
        super(props)
        this.state = {
            dropdown: false,
            language: 'English',
            timezone: '(UTC+00:00) El Aaiun',
            currency: 'USD ($)',
            profileVisibility: 'Everyone',
            message: 'Everyone',
            category: 'Enable',
            _id: this.props.user._id
        }
    }

    async componentWillMount() {
        await this.props.getPreference(this.props.user._id)
        const {language, timezone, currency, profile_visibility, message, category} = this.props.preference
        if(this.props.havePreference){
            await this.setState({
                language, 
                timezone, 
                currency, 
                profileVisibility: profile_visibility, 
                message, 
                category
            })
        }
    }

    async resetPreference() {
        await this.props.deletePreference(this.props.preference._id)
        const {language, timezone, currency, profileVisibility, message, category} = this.props.preference
        await this.setState({
            language, 
            timezone, 
            currency, 
            profileVisibility, 
            message, 
            category
        })
        alert('Deleted')
    }

    setLanguage(data) {
        this.setState({language: data})
    }

    setCurrency(data) {
        this.setState({currency: data})
    }

    setTimezone(data) {
        this.setState({timezone: data})
    }

    render() {
        return (
            <div>
                <TopBar/>
                <OuterContainer>
                    <OuterLeftDiv>
                        <LeftMenu nowPath={this.props.location.pathname}/>
                    </OuterLeftDiv>
                    <Container>
                        <div style={{fontWeight: 'bold', marginTop: 20, marginLeft: '2.3%', color: '#515C67'}}>Edit Preferences</div>
                        <SepSection/>
                        <SectionDiv>
                            <Section left color={'#6F727D'}>Localization</Section>
                            <Section>
                                <Text>Language</Text>
                                <Dropdown displayValue={this.state.language} value={languages} setData={this.setLanguage.bind(this)}/>
                                <Text em={0.8} color={'#C2C4CB'}  weight={'regular'}>Interested in helping translate Fancy? <span style={{color: '#527FB7'}}>Let us know.</span></Text>
                                <br/>
                                <Text>Time zone</Text>
                                <Dropdown displayValue={this.state.timezone} value={timezones} setData={this.setTimezone.bind(this)}/>
                                <Text>Currency</Text>
                                <Dropdown displayValue={this.state.currency} value={currencies} setData={this.setCurrency.bind(this)}/>
                            </Section>
                        </SectionDiv>
                            <SepSection/>
                        <SectionDiv>
                            <Section left color={'#6F727D'}>Privacy</Section>
                            <Section>
                                <Text>Profile visibility</Text> 
                                <Text em={0.8} color={'#C2C4CB'} weight={'regular'}>Manage who can see your activity, things you fancy, your followers, people you follow or in anyone's search results.</Text>
                                <RadioDiv>
                                    <input type={'radio'} name={'profile_vis'} checked={this.state.profileVisibility==='Everyone'?true:false} onChange={data => {this.setState({profileVisibility: data.target.value})}} value={'Everyone'} style={{margin: '0 7px 0 0'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}>Everyone</Text>
                                    <input type={'radio'} name={'profile_vis'} checked={this.state.profileVisibility==='Private'?true:false} onChange={data => {this.setState({profileVisibility: data.target.value})}} value={'Private'} style={{margin: '0 7px 0 20px'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}><FaLock/>Private</Text>
                                </RadioDiv>
                                <Text>Messages</Text>  
                                <Text em={0.8} color={'#C2C4CB'} weight={'regular'}>Control who can send you messages.</Text>
                                <RadioDiv>
                                    <input type={'radio'} name={'message'} checked={this.state.message==='Everyone'?true:false} onChange={data => {this.setState({message: data.target.value})}} value={'Everyone'} style={{margin: '0 7px 0 0'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}>Everyone</Text>
                                    <input type={'radio'} name={'message'} checked={this.state.message==='People you follow'?true:false} onChange={data => {this.setState({message: data.target.value})}} value={'People you follow'} style={{margin: '0 7px 0 20px'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}>People you follow</Text>
                                    <input type={'radio'} name={'message'} checked={this.state.message==='No one'?true:false} onChange={data => {this.setState({message: data.target.value})}} value={'No one'} style={{margin: '0 7px 0 20px'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}><FaLock/>No one</Text>
                                </RadioDiv>
                                <Text>Recently viewed</Text>  
                                <Text em={0.8} color={'#C2C4CB'} weight={'regular'}>Manage your Fancy browsing history.</Text>
                                <Text color={'#527FB7'}>Delete all items</Text>
                            </Section>
                        </SectionDiv>
                            <SepSection/>
                        <SectionDiv>
                            <Section left color={'#6F727D'}>Content</Section>
                            <Section>
                                <Text>Category lists</Text>  
                                <Text em={0.8} color={'#C2C4CB'} weight={'regular'}>Automatically add Fancy'd items to the Category list</Text>
                                <RadioDiv>
                                    <input type={'radio'} name={'category'} checked={this.state.category==='Enable'?true:false} onChange={data => {this.setState({category: data.target.value})}} value={'Enable'}  style={{margin: '0 7px 0 0'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}>Enable</Text>
                                    <input type={'radio'} name={'category'} checked={this.state.category==='Disable'?true:false} onChange={data => {this.setState({category: data.target.value})}} value={'Disable'} style={{margin: '0 7px 0 20px'}}/><Text em={0.8} color={'#6F727D'}  weight={'regular'}>Disable</Text>
                                </RadioDiv>
                            </Section>
                        </SectionDiv>
                        <SepSection/>
                        <SepSection/>
                        <div style={{display: 'flex', flex: 1, justifyContent: 'flex-end', margin: '0 15px 15px 0'}}>
                            {this.props.havePreference?
                                <div>
                                    <Button fontColor={'white'} color={'red'} onClick={() => this.resetPreference()}>Delete Preferences</Button>
                                    <Button onClick={() => {this.props.updatePreference(this.state);alert('Saved')}}>Edit Preferences</Button>
                                </div>
                            :
                                <Button onClick={() => {this.props.createPreference(this.state);alert('Saved')}}>Save Preferences</Button>
                            }
                        </div>
                    </Container>
                </OuterContainer>
            </div>
        )
    }
}

const OuterContainer = styled.div`
    display: flex;
    flex: 1;
    justify-content: center;
    margin-top: 15px;
    @media (max-width: 750px) {
        flex-direction: column;
    }
`

const Container = styled.div`
    border-radius: 2px;
    width: 60%;
    background-color: white;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
    margin-left: 20px;
    margin-bottom: 20px;
    @media (max-width: 750px) {
        width: 90%;
    }
`

const SepSection = styled.div`
    margin: 15px 0 15px 0;
    border-bottom: 1px solid #ddd;
`

const SectionDiv = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-around;
`

const Section = styled.div`
    color: ${props => props.color || 'black'};
    width: ${props => props.left?'25%':'65%'};
`

const Text = styled.div`
    font-size: ${props => props.em || 0.9}em;
    font-weight: ${props => props.weight || 'bold'};
    color: ${props => props.color || '#515C67'};
`
const Button = styled.button`
    height: 34px;
    weught: 190px;
    border-radius: 4px;
    border: 1px solid #515C67;
    color: ${props => props.fontColor || '#515C67'};
    font-size: 0.8em;
    font-family: 'Hind';
    font-weight: bold;
    background-color: ${props => props.color || '#F8F8F8'}
`

const RadioDiv = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    margin: 15px 0 15px 0;
`

const OuterLeftDiv = styled.div`
    width: 16%;
    @media (max-width: 750px) {
        margin-left: 20px;
        margin-bottom: 20px;
        width: 90%;
    }
`

const mapStateToProps = state => {
    return { 
        user: state.auth.get('user'),
        preference: state.preference.get('preference'), 
        havePreference: state.preference.get('havePreference') 
    }
}

export default connect(mapStateToProps, { createPreference, getPreference, deletePreference, updatePreference })(EditPreferencesComponent)