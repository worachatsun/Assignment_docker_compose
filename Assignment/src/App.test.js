import React from 'react'
import { expect } from 'chai'
import { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Dropdown from '../src/commons/Dropdown'
import configureStore from 'redux-mock-store'
import * as actions from './actions'
import {
  BrowserRouter as Router
} from 'react-router-dom'
import { Provider, connect } from 'react-redux'
import SigninPage from '../src/components/Authentication/SigninPage'
import SignupPage from '../src/components/Authentication/SignupPage'
import EditPreferencesComponent from '../src/components/EditPreferencesComponent'
import TopBar from '../src/commons/TopBar'
import LeftMenu from '../src/commons/LeftMenu'
import { languages, timezones, currencies } from './assets/resource'

describe('Testing ReassignLocationMenu component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Dropdown displayValue={timezones[0]} value={timezones} setData={jest.fn()} />
    )
    expect(wrapper.find('div').length).to.equal(1)
  })
})

describe('TopBar Component', () => {
  const initialState = {}

  const mockStore = configureStore()
  const store = mockStore(initialState)
  it('should have search input', () => {
    expect(mount(<TopBar store={store} />).find('input[placeholder="Search Fancy"]').length).to.equal(1)
  })
})

describe('SigninPage Component', () => {
  const initialState = {
    user: {},
    isLoggedIn: false
  }

  const mockStore = configureStore()
  const store = mockStore(initialState)

  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <SigninPage />
      </Router>
    </Provider>
  )

  it('has a username input field', () => {
    expect(wrapper.containsMatchingElement(<input type="text" />)).to.be.true
  })

  it('has a password input field', () => {
    expect(wrapper.containsMatchingElement(<input type="password" />)).to.be.true
  })
})

describe('SignupPage Component', () => {
  const initialState = {
    user: {},
    isLoggedIn: false
  }

  const mockStore = configureStore()
  const store = mockStore(initialState)

  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <SignupPage />
      </Router>
    </Provider>
  )

  it('has a username input field', () => {
    expect(wrapper.containsMatchingElement(<input type="text" />)).to.be.true
  })

  it('has a email input field', () => {
    expect(wrapper.containsMatchingElement(<input type="email" />)).to.be.true
  })

  it('has a password input field', () => {
    expect(wrapper.containsMatchingElement(<input type="password" />)).to.be.true
  })
})