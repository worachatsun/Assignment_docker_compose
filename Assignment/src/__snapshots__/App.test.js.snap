// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ReassignLocationMenu component renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    displayValue="(UTC+00:00) El Aaiun"
    setData={[Function]}
    value={
        Array [
            "(UTC+00:00) El Aaiun",
            "(UTC+00:00) Faroe",
            "(UTC+00:00) Freetown",
            "(UTC+00:00) Guernsey",
            "(UTC+00:00) Isle of Man",
            "(UTC+00:00) Jersey",
            "(UTC+00:00) Lisbon",
            "(UTC+00:00) Lome",
            "(UTC+00:00) London",
            "(UTC+00:00) Madeira",
            "(UTC+00:00) Monrovia",
            "(UTC+00:00) Nouakchott",
            "(UTC+00:00) Ouagadougou",
            "(UTC+00:00) Reykjavik",
            "(UTC+00:00) Sao Tome",
            "(UTC+00:00) St. Helena",
            "(UTC+00:00) UTC",
            "(UTC+01:00) Algiers",
            "(UTC+01:00) Amsterdam",
            "(UTC+01:00) Andorra",
            "(UTC+01:00) Bangui",
            "(UTC+01:00) Belgrade",
            "(UTC+01:00) Berlin",
            "(UTC+01:00) Bratislava",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          onClick={[Function]}
>
          <styled.div>
                    (UTC+00:00) El Aaiun
                    <FaCaretDown />
          </styled.div>
</div>,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <styled.div>
            (UTC+00:00) El Aaiun
            <FaCaretDown />
</styled.div>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "(UTC+00:00) El Aaiun",
              <FaCaretDown />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "(UTC+00:00) El Aaiun",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            onClick={[Function]}
>
            <styled.div>
                        (UTC+00:00) El Aaiun
                        <FaCaretDown />
            </styled.div>
</div>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <styled.div>
              (UTC+00:00) El Aaiun
              <FaCaretDown />
</styled.div>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "(UTC+00:00) El Aaiun",
                <FaCaretDown />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "(UTC+00:00) El Aaiun",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Testing ReassignLocationMenu component renders as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    count={1}
    displayValue="(UTC+00:00) El Aaiun"
    setData={[Function]}
    value={
        Array [
            "(UTC+00:00) El Aaiun",
            "(UTC+00:00) Faroe",
            "(UTC+00:00) Freetown",
            "(UTC+00:00) Guernsey",
            "(UTC+00:00) Isle of Man",
            "(UTC+00:00) Jersey",
            "(UTC+00:00) Lisbon",
            "(UTC+00:00) Lome",
            "(UTC+00:00) London",
            "(UTC+00:00) Madeira",
            "(UTC+00:00) Monrovia",
            "(UTC+00:00) Nouakchott",
            "(UTC+00:00) Ouagadougou",
            "(UTC+00:00) Reykjavik",
            "(UTC+00:00) Sao Tome",
            "(UTC+00:00) St. Helena",
            "(UTC+00:00) UTC",
            "(UTC+01:00) Algiers",
            "(UTC+01:00) Amsterdam",
            "(UTC+01:00) Andorra",
            "(UTC+01:00) Bangui",
            "(UTC+01:00) Belgrade",
            "(UTC+01:00) Berlin",
            "(UTC+01:00) Bratislava",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          onClick={[Function]}
>
          <styled.div>
                    (UTC+00:00) El Aaiun
                    <FaCaretDown />
          </styled.div>
</div>,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <styled.div>
            (UTC+00:00) El Aaiun
            <FaCaretDown />
</styled.div>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "(UTC+00:00) El Aaiun",
              <FaCaretDown />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "(UTC+00:00) El Aaiun",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            onClick={[Function]}
>
            <styled.div>
                        (UTC+00:00) El Aaiun
                        <FaCaretDown />
            </styled.div>
</div>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <styled.div>
              (UTC+00:00) El Aaiun
              <FaCaretDown />
</styled.div>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "(UTC+00:00) El Aaiun",
                <FaCaretDown />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "(UTC+00:00) El Aaiun",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
